### PR TITLE
fix: special chars in block titles

### PR DESCRIPTION
**Pull request by nicorikken**
This issue will fix special characters breaking block title highlighting.

### CODE REVIEW

File: grammars/asciidoc.cson [hunk 1]
```diff
@@ -48,8 +48,7 @@
   # ========================
   #
   {
-    'begin': '^\\.(\\w[\\w ]*\\w)'
-    'end': '$'
+    'match': '(?<=^\\.)(\\w.+)(?=$)'
```
Comment (mojavelinux) [inline grammars/asciidoc.cson]: There is another PR also proposing to fix this issue. See #48.

In that PR, I describe the correct way to match a block title.

```
'begin': '(?<=^\\.)([^\\s.].*)'
```

In other words, it should match a title that is at least one character in length and does not start with a space or dot.

**Comment by nicorikken** [comment]
Based on the asciidoctor.rb source, and taking in my learnings in the italic highligh ( #21 ), I've now come up with two matches, for both constrained and unconstrained phrases:

Constrained: `'match': '(^|[^\\w;:}])(?:\\[([^\\]]+?)\\])?(?!\\*{3})(\\*\\S\\*|\\*\\S[\\s\\S]*?\\S\\*)(?![\\w])'`
Unconstrained: `'match': '\\\\?(?:\\[([^\\]]+?)\\])?(\\*\\*[\\s\\S]+?\\*\\*)'`

It needs some more testing and unit tests, but for now this seems to be the right solution.

**Comment by mojavelinux** [comment]
Excellent! However, I think your comment applies to #21, not this one.

@nicorikken You've been doing excellent work on this package and we'd like to invite you to become a "committer". That will give you the authority to merge your pull requests once we decide they're ready to go, saving you the trouble of having to wait on anyone else to merge.

**Comment by mojavelinux** [comment]
I think we should merge #48 when it is ready, then rebase this one and proceed to include the test and perhaps a refinement to the regex.

**Comment by nicorikken** [comment]
@mojavelinux Thank you very much for the invite. I'll happily help to handle the PRs. I'll probably need some guidance and help in handling especially the first few PRs, to get familiar with Asciidoctor practices and the ins and outs of git merging in general. But I'll get back to you on that then.

**Comment by mojavelinux** [comment]
Excellent! Of course, we are here to help you! The most important thing is that we start to integrate your great work.

**Comment by nicorikken** [comment]
#48 is merged, and now indeed tests can be included. For now I have no reason to doubt the included regexes.

**Comment by mojavelinux** [comment]
I think this PR needs to be rebased. Then we can assess where we stand.

**Comment by nicorikken** [comment]
@mojavelinux do you agree on merging this changed test-case?

**Comment by mojavelinux** [comment]
Yes indeed!

Technically, we could consolidate the rule so that it only uses the begin key:

```
  {
    'begin': '^\\.([^\\s.].*)$'
    'captures':
      '1': 'name': 'markup.heading.blocktitle.asciidoc'
  }
```

But that's effectively the same as what it is now. It's up to you if you'd like to consolidate it.

**Comment by nicorikken** [comment]
The consolidation scrambles up my parsing. I guess a match would do likewise. I'll leave it as merged for now.

**Comment by nicorikken** [comment]
Merged!

**Comment by mojavelinux** [comment]
Thanks for checking!

:+1:

**Pull request merged**